src/pipecat/pipeline/task.py

```diff
@@ -552,8 +575,16 @@ async def _process_down_queue(self):
             if isinstance(frame, self._reached_downstream_types):
                 await self._call_event_handler("on_frame_reached_downstream", frame)
 
-            if isinstance(frame, (EndFrame, StopFrame)):
+            if isinstance(frame, StartFrame):
+                await self._call_event_handler("on_pipeline_started", frame)
```

Comment:
If I'm reading the code correctly, this event is called after the StartFrame has passed all the way through the pipeline. So when you get this event, all the processors in the pipeline have already seen the StartFrame. Is that right?

Reply:
Exactly. We have two hidden processors at the beginning and end of a pipeçine that allow us to do this.